### PR TITLE
REF/TST: Add more pytest idiom to parsers tests

### PR DESCRIPTION
**Pull request by gfyoung**
Adds more idiom to the following:

* `parser/skiprows` --> `parser/test_skiprows`
* `parser/na_values` --> `parser/test_na_values`
* `parser/parse_dates` -->` parser/test_parse_dates`
* `parser/multithread` --> `parser/test_multi_thread`

Also adds new test suite: `test_date_converters` (was previously hidden in `parser/parse_dates`)

Builds off of #23712.

### CODE REVIEW

File: pandas/tests/io/parser/test_parse_dates.py [hunk 1]
```diff
@@ -0,0 +1,835 @@
...
+    index.name = "date"
+
+    expected = DataFrame({"A": ["a", "b", "c"], "B": [1, 3, 4],
+                          "C": [2, 4, 5]}, index=index)
+    tm.assert_frame_equal(result, expected)
+
+
+# Bug in https://github.com/dateutil/dateutil/issues/217
+# has been addressed, but we just don't pass in the `yearfirst`
+@pytest.mark.skip(reason="yearfirst is not surfaced in read_*")
```
Comment (gfyoung) [inline pandas/tests/io/parser/test_parse_dates.py]: By not surfacing `yearfirst`, we implicitly always have `yearfirst=False` (we want `yearfirst=True` here)

Question is: do we want to surface this parameter to `read_csv` (adds functionality but adds more bloat to API)?  Or, should we go the other way and deprecate `dayfirst` and require users to pass in a parser to the `date_parser` argument (reduces bloat but will be a breaking change for users)?

File: pandas/tests/io/parser/test_parse_dates.py [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+        [datetime(1999, 1, 27, 23, 0), datetime(1999, 1, 27, 22, 56),
+         "KORD", "19990127", " 23:00:00", " 22:56:00",
+         -0.59, 1.71, 4.6, 0.0, 280.0],
+    ], columns=["nominal", "actual", "X0", "X1", "X2",
+                "X3", "X4", "X5", "X6", "X7", "X8"])
+
+    if not keep_date_col:
+        expected = expected.drop(["X1", "X2", "X3"], axis=1)
+    elif parser.engine == "python":
+        expected["X1"] = expected["X1"].astype(np.int64)
```
Comment (gfyoung) [inline pandas/tests/io/parser/test_parse_dates.py]: So this is an inconsistency in how we have columns that are aggregated to create one date column.  The C engine does not continue to parse them and infer dtype, whereas the Python does parse and infer dtypes on these columns as if they were regular data columns.

Which engine is more correct?

File: pandas/tests/io/parser/test_parse_dates.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/io/parser/test_parse_dates.py]: don't skip, rather xfail

File: pandas/tests/io/parser/test_parse_dates.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/io/parser/test_parse_dates.py]: can you open an issue here for followup

File: pandas/tests/io/parser/test_parse_dates.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tests/io/parser/test_parse_dates.py]: Done.  I also see that #2586 is already discussing this.

File: pandas/tests/io/parser/test_parse_dates.py (hunk 2 above)
Comment (gfyoung) [reply pandas/tests/io/parser/test_parse_dates.py]: Opened #23845

**Comment by jreback** [comment]
thanks @gfyoung

**Pull request merged**